### ISSUE TITLE
[Arch Linux] steam.sh: line 813: 15086 Segmentation fault

### ISSUE THREAD

**Issue by ndavd** [ValveSoftware/steam-for-linux#10841]
#### Your system information

* Steam client version (build number or date): `multilib/steam 1.0.0.79-1` 
* Distribution (e.g. Ubuntu): Arch
* Opted into Steam client beta?: Yes
* Have you checked for system updates?: Yes
* Steam Logs: [steam-logs.tar.gz](https://github.com/ValveSoftware/steam-for-linux/files/15190203/steam-logs.tar.gz)
* GPU: Nvidia

```
~[main]₿ steam
steam.sh[14959]: Running Steam on arch rolling 64-bit
steam.sh[14959]: STEAM_RUNTIME is enabled automatically
setup.sh[15032]: Steam runtime environment up-to-date!
steam.sh[14959]: Steam client's requirements are satisfied
[2024-05-02 16:01:35] Startup - updater built Apr 30 2024 22:32:46
[2024-05-02 16:01:35] Startup - Steam Client launched with: '/home/ndavd/.local/share/Steam/ubuntu12_32/steam'
[2024-05-02 16:01:35] Opted in to client beta 'publicbeta' via beta file
You are in the 'publicbeta' client beta.
05/02 16:01:35 /tmp/dumps: insufficient permissions - delete and recreate
05/02 16:01:35 minidumps folder is set to /tmp/dumps
05/02 16:01:35 Init: Installing breakpad exception handler for appid(steam)/version(1714520927)/tid(15086)
crash_20240502160135_2.dmp[15089]: Uploading dump (out-of-process)
/tmp/dumps/crash_20240502160135_2.dmp
/home/ndavd/.local/share/Steam/steam.sh: line 813: 15086 Segmentation fault      (core dumped) "$STEAMROOT/$STEAMEXEPATH" "$@"
~[main]₿ crash_20240502160135_2.dmp[15089]: Finished uploading minidump (out-of-process): success = yes
crash_20240502160135_2.dmp[15089]: response: CrashID=bp-4dd0d164-4f8c-4733-b725-03f182240502
crash_20240502160135_2.dmp[15089]: file ''/tmp/dumps/crash_20240502160135_2.dmp'', upload yes: ''CrashID=bp-4dd0d164-4f8c-4733-b725-03f182240502''```

**Comment by jlsalvador** [comment]
running through gdb seems it doesn't crash.

```
$ gdb steam
(gdb) run
[...]
(gdb) quit
```

🤔

**Comment by ndavd** [comment]
> running through gdb seems it doesn't crash.
> 
> ```
> $ DEBUGGER=gdb steam
> (gdb) run
> [...]
> (gdb) quit
> ```
> 
> 🤔

Can confirm, yea. That's very weird, no?

**Comment by jlsalvador** [comment]
It's because `lib32-glibc` 2.39-3. Downgrade it to 2.39-2 and steam will not crash any more.

```
$ sudo pacman -U https://archive.archlinux.org/packages/l/lib32-glibc/lib32-glibc-2.39-2-x86_64.pkg.tar.zst
```

**Comment by kisak-valve** [comment]
Hello @jlsalvador, the distro packaging regression should be mentioned to your distro's package maintainers.

**Comment by jlsalvador** [comment]
> Hello @jlsalvador, the distro packaging regression should be mentioned to your distro's package maintainers.

Yes I known; I'm on it (I'm trying to create an issue for Archlinux maintainers). My message was to help other users with the same problem. Thank you @kisak-valve 👍.

**Comment by DataCpt** [comment]
> It's because `lib32-glibc` 2.39-3. Downgrade it to 2.39-2 and steam will not crash any more.
> 
> ```
> $ sudo pacman -U https://archive.archlinux.org/packages/l/lib32-glibc/lib32-glibc-2.39-2-x86_64.pkg.tar.zst
> ```

Bless you, that was a stressful hour. How could you tell?

**Comment by ptr1337** [comment]
Added Packaging Issue:
https://gitlab.archlinux.org/archlinux/packaging/packages/glibc/-/issues/10

and next time, you should create a upstream packaging issue, if you know that downgrading fixes it.
It does really not help anyone, if the issue is for you fixed but rolled out to many people.

**Comment by jlsalvador** [comment]
> Added Packaging Issue: https://gitlab.archlinux.org/archlinux/packaging/packages/glibc/-/issues/10
> 
> and next time, you should create a upstream packaging issue, if you know that downgrading fixes it. It does really not help anyone, if the issue is for you fixed but rolled out to many people.

Thank you @ptr1337 for reporting the issue in their Gitlab. Just for your context, I reported this issue through emailing directly to Frederik Schwan as soon as I known it is a packaging regression. I can not create an Archlinux Gitlab account (because its Gitlab registration is locked); and because of that I was not be able to create this issue in their Gitlab.

Thank you all.

**Comment by ptr1337** [comment]
@jlsalvador @ndavd 

Frederik is requesting more logs from you, I can not reproduce the issue on my machine.
Could you please update to -3 again and provide the maintainer the required logs?

Thanks!
https://gitlab.archlinux.org/archlinux/packaging/packages/glibc/-/issues/10#note_182195

**Comment by freswa** [comment]
Is there anyone affected who _doesn't_ run nvidia drivers?

**Comment by HydeFromT70s** [comment]
@freswa I've sent on your @ requested log + coredump (don't have acc on GitLab).
I hope it will help.

**Comment by ndavd** [comment]
> @jlsalvador @ndavd
> 
> Frederik is requesting more logs from you, I can not reproduce the issue on my machine. Could you please update to -3 again and provide the maintainer the required logs?
> 
> Thanks! https://gitlab.archlinux.org/archlinux/packaging/packages/glibc/-/issues/10#note_182195

Sure, I don't have an account there, could you share it with him? [steam-runtime-system-info.json](https://github.com/ValveSoftware/steam-for-linux/files/15193840/steam-runtime-system-info.json)

**Comment by Paiusco** [comment]
@freswa also emailed you my logs even tho I also use nvidia drivers. Hope it helps to find a pattern.
(requested an account into gitlab as well)

**Comment by jlsalvador** [comment]
@freswa I can not reply you through archlinux gitlab, but I can here :stuck_out_tongue_winking_eye:. About your last comment (note_182230) about removing `/usr/lib32/gconv/gconv-modules.cache`, I'm sorry,  this doesn't fix the crash.

**Comment by mries92** [comment]
> It's because `lib32-glibc` 2.39-3. Downgrade it to 2.39-2 and steam will not crash any more.
> 
> ```
> $ sudo pacman -U https://archive.archlinux.org/packages/l/lib32-glibc/lib32-glibc-2.39-2-x86_64.pkg.tar.zst
> ```

Thank you so much! Like DataCpt I am also curious about how you figured that out. I don't see anything about this library in the crash logs.

**Comment by nipnipj** [comment]
This happened on my second day using arch. Hours of panic.

**Comment by srmfx** [comment]
> This happened on my second day using arch. Hours of panic.

You got really lucky this time,

**Comment by freswa** [comment]
Please test `glibc-2.39-4`

**Comment by iddm** [comment]
> Please test `glibc-2.39-4`

Works.

**Comment by HydeFromT70s** [comment]
> Please test `glibc-2.39-4`.

Can confirm that Steam works again with glibc-2.39-4.
Thanks for solving the problem quickly.

**Comment by ndavd** [comment]
Legendary🪙

**Comment by nipnipj** [comment]
I think my issue started after I used `sudo pacman -Syu`, which makes me wonder whether this is the best way to keep up-to-date the system.
I used this command today and steam didnt open. After I installed `glibc-2.39-4` and it is working now.

**Comment by ptr1337** [comment]
> I think my issue started after I used `sudo pacman -Syu`, which makes me wonder whether this is the best way to keep up-to-date the system. I used this command today and steam didnt open. After I installed `glibc-2.39-4` and it is working now.

Just use -Syu. This regression happened due a change in buildflags and arch did miss to not apply this to lib32 packages.
Things like this can happen, when doing changes on the buildsystem and updates.

**Comment by TTimo** [comment]
Closing per Arch fix (glibc-2.39-4)
